### DISCUSSION TITLE
Is there a way to re-compute answers to computed questions on copier update?

### DISCUSSION THREAD

**Discussion by AgentK9** [copier-org/copier#1666] [Q&A] [answered]
I wasn't sure if this was already implemented, so decided to make a discussion instead of an issue.
Say you're updating a template and specify a changed answer using `--data=x=2` (`x` was previously 1), and `y` defines its default value as `{{x + 1}}`. Is there a way to tell copier to recompute `y` while using `--defaults`? Does it already do it automatically? Note that I don't want to go through the questionnaire again.

**Comment by sisp** [comment, marked as answer]
Unless `y` is a `when: false` question, so it's a computed value and no question will be ever be asked, it isn't possible at the moment. A recorded answer has a higher priority than using `--defaults`.

Would you mind sharing your use case?

**Reply by AgentK9** [reply]
I think that the computed questions is what I was getting at - it wasn't clear to me that they weren't stored until I actually started using the tool. Thanks!
